gamedat.cc

```diff
@@ -482,7 +482,7 @@ void Game_window::write_saveinfo() {
 
 			char name[18];
 			std::string namestr = npc->get_npc_name();
-			strncpy(name, namestr.c_str(), 18);
+			memcpy(name, namestr.c_str(), 18);
```

Comment:
Ack, no. Undefined behavior here whenever `namestr.size() < 17`.  A better solution is to do away with `name` variable, replace the `strncpy` call by `namestr.resize(18, '\0')` , and use `namestr.c_str()` instead of `name` in the call to write.